lib/characteristic.dart

```diff
@@ -68,12 +68,23 @@ mixin WithValue on Characteristic {
   Uint8List value;
 }
 
-class CharacteristicWithValue extends Characteristic with WithValue {
-  CharacteristicWithValue.fromJson(
+mixin WitTransactionId on Characteristic {
+  String transactionId;
+}
+
+class CharacteristicWithValueWithTransactionId extends Characteristic with WithValue, WitTransactionId {
```

Comment:
`CharacteristicWithValue` and `CharacteristicWithValueAndTransactionId`? One with transactionId is only need for monitoring, and then the mixin could be on `CharacteristicWithValue`.

Reply:
Actually we might not need mixins at all, since we're declaring classes with them in the same file and they're not holding any logic.